modules/transport-netty4/build.gradle

```diff
@@ -0,0 +1,168 @@
+import com.sun.org.apache.bcel.internal.util.ClassPath
+
+import java.util.logging.LogManager
+import java.util.zip.Deflater
+import java.util.zip.Inflater
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ TODOs:
+   * fix permissions such that only netty4 can open sockets etc?
+   * fix the hack in the build framework that copies transport-netty4 into the integ test cluster
+   * maybe figure out a way to run all tests from core with netty4/network?
+ */
+esplugin {
+    description 'Netty 4 based transport implementation'
+    classname 'org.elasticsearch.transport.Netty4Plugin'
+    hasClientJar = true
+}
+
+compileTestJava.options.compilerArgs << "-Xlint:-cast,-deprecation,-rawtypes,-try,-unchecked"
+
+dependencies {
+    // network stack
+    compile "io.netty:netty-buffer:4.1.4.Final-SNAPSHOT"
```

Comment:
I wonder if this is a blocker, if we run against arbitrary snapshots we are subject to fail on upstream problems, shall we do that or should we, for the time being use some defined snapshot with a build ID deployed to S3 or something?